### ISSUE TITLE
Inferred return type of a decorated function is incorrect

### ISSUE THREAD

**Issue by michitaro** [microsoft/python-language-server#1852]
Inferred return type of a decorated function is incorrect.
(That is the one for the function without the decoration.)

## Environment data

- Language Server version: 0.5.10.0
- OS and version: macOS 10.15
- Python version (& distribution if applicable, e.g. Anaconda): Anaconda3 Python 3.7.3

## Expected behaviour

- The LS infers a correct return type of a decorated function.

## Actual behaviour

- The LS infers a return type of a decorated function as if the function is not decorated.

## Logs

N/A

## Code Snippet / Additional lnformation

```python
from typing import Any, Callable


def stringify(f: Callable[..., Any]) -> Callable[..., str]:
    def wrapper(*args) -> str:
        return str(f(*args))
    return wrapper

@stringify
def one() -> int:
    return 1

value = one() # inferred type of `value` by the LS is `int`,
              # but it should be `str`

print(type(value))
# <class 'str'>
```

![Screen Shot 2019-12-25 at 16 52 50](https://user-images.githubusercontent.com/3928908/71438125-d0eb7d80-2737-11ea-98b2-98acf36e58fe.png)

**Comment by MikhailArkhipov** [comment]
True, currently analysis does not look into decorators in order to find out possible return type. Also, `Callable` is not currently handled. #1115 and #535
